module2/README.md

```diff
@@ -0,0 +1,97 @@
+
+b) What is Continuous Learning, and why is it important in mobile development?
+
+Working developers engage in continuous learning to update and expand their skillset over the course of their career. The mobile development echosystem evolves quickly, so it is important to continuously learn to understand the latest tools and technologies.
+
+c) How can you find out what modifiers a View has?
+
+Select it in the SwiftUI preview panel, open the details inspector, and select the attributes inspector. The modifiers will be listed there.
+
+d) What is a breakpoint?
+
+It is a point in code where the debugger will stop execution and show all the current values of all the current relevant variables.
+
+e) How can you access environment values in your App?
+
+Using the @Environment(.[keypath]) directive with the correct keypath for the environment variable you want.
+
+f) How can you determine, in code, if the App is in Dark or Light Mode?
+
+@Environment(\.colorScheme)
+
+g) Why are magic numbers an issue, and how should you avoid them?
+
+Magic numbers are configuration values sprinked line throughout code. They can be duplicative and hard to locate. Turning these numbers into static constants keeps them all in one place where they are easy to edit and duplciates are easy to spot and remove.
+
+h) How can you view your App in Light and Dark Modes simultaneously?
+
+Setup multiple preview views with .preferredColorScheme set to .light and .dark 
+
+i) Below is an image of the Canvas from Xcode. The Canvas is in selectable mode. Can you explain why the red background does not cover the entire button area?
+
+It is because a padding modifier is being appled to the view after the red background. The red-backbground modifier returns a new view which is the same size as the original view, but with a red-background. Applying a padding modifier after this returns a new view with white padding around the view with the red background.
+
+j) Modifier padding(10) adds padding to the view's top, bottom, left, and right sides. How could a padding of 10 be added to only the left and right sides of the view? The answer for this question should be a short section of code.
+
+Use the modifier .padding(.horizontal, 10)
+
+k) Provide two reasons why you would want to extract views.
+
+One - to re-use the views as subviews in multiple parent views
+Two - to keep the code for views from being too long and hard to understand
+
+l) How can you determine, in code, if the device is in Portrait or Landscape mode?
+
+You can approximate this with the environment variables verticalSizeClass and horizontalSizeClass. If vertical is compact and horizontal is regular, then you are in Landscape. If vertical is regular and horizontal is compact, then you are in Portrait. This would work for phone style devices. 
```

Comment:
Its much trickier on iPad devices. You frequently have to resort to using a `GeometryReader` modifier and checking to see if some view is wider than it is long or longer than it is wide. A lot of people will set up their own `@Environment()` value to help with this.